File: problems/1854-maximum-population-year/solution.c
int maximumPopulation(int** logs, int logsSize, int* logsColSize) {
    int years[101] = {0};
    for (int i = 0; i < logsSize; i++) {
        for (int year = logs[i][0]; year < logs[i][1]; year++) {
            years[year - 1950]++;
        }
    }
    int maxPop = 0;
    int maxYear = 1950;
    for (int i = 0; i < 101; i++) {
        if (years[i] > maxPop) {
            maxPop = years[i];
            maxYear = 1950 + i;
        }
    }
    return maxYear;
}

File: problems/1854-maximum-population-year/solution.java
class Solution {
    public int maximumPopulation(int[][] logs) {
        int[] years = new int[101];
        for (int[] log : logs) {
            years[log[0] - 1950]++;
            years[log[1] - 1950]--;
        }
        int maxPop = 0;
        int maxYear = 1950;
        int currentPop = 0;
        for (int i = 0; i < 101; i++) {
            currentPop += years[i];
            if (currentPop > maxPop) {
                maxPop = currentPop;
                maxYear = 1950 + i;
            }
        }
        return maxYear;
    }
}